package/MDAnalysis/analysis/rms.py

```diff
@@ -484,6 +485,63 @@ def save(self, filename=None):
         return filename
 
 
+def rmsf(atomgroup, start=0, stop=-1, step=1, progout=None):
+    """Calculate RMSF of given atoms across a trajectory.
+
+    This function does no fitting; it uses the coordinates of the atoms as-is.
+
+    This function implements an algorithm for computing sums of squares while
+    avoiding overflows and underflows; please reference:
+
+    .. [Welford1962] B. P. Welford (1962)
+       "Note on a Method for Calculating Corrected Sums of Squares and 
+            Products." Technometrics  4(3):419-420.
+
+    :Arguments:
+        *atomgroup*
+            AtomGroup to obtain RMSF for
+        *start*
+            starting frame [0]
+        *stop*
+            stopping frame [-1]
+        *step*
+            step between frames [1]
+        *progout*
+            number of frames to iterate through between updates to progress
+            output; ``None`` for no updates [``None``]
+
+    :Returns:
+        *rmsf*
+           :class:`numpy.ndarray` with length equal to *atomgroup*, giving the
+           RMSF (Å) for each atom
+
+    """
+    sumsquares = numpy.zeros((atomgroup.numberOfAtoms(), 3))
+    means = numpy.array(sumsquares)
+
+    # set up progress output
+    if progout:
+        percentage = ProgressMeter(atomgroup.universe.trajectory.numframes, 
+                                   interval=progout)
+    else:
+        percentage = ProgressMeter(atomgroup.universe.trajectory.numframes, 
+                                   quiet=True)
+
+    for k, ts in enumerate(atomgroup.universe.trajectory[start:stop:step]):
+        sumsquares = sumsquares + (k * (atomgroup.positions - means)**2)/(k + 1)
```

Comment:
Good suggestion; I've updated the loop with a couple changes that should decrease the number of temporaries. The *best* solution would be to use numexpr, but I didn't want to add that dependency to mdanalysis without some discussion first.